components/UI/ContactDetails.js

```diff
@@ -13,22 +13,23 @@ const ContactDetails = (props) => {
   const [toggleDetails, setToggleDetails] = useState(false);
 
   //Find the profile which matches the id we passed on clicking to the detail
-  const profilesArray = useSelector((state) => state.profiles.allProfiles).filter(
-    (profile) => profile.profileId === props.profileId
+  let profile = useSelector((state) =>
+    state.profiles.allProfiles.find(({ profileId }) => profileId === props.profileId)
   );
 
-  let objectForDetails = profilesArray[0];
+  if (!profile) {
+    return null;
+  }
 
-  const contactEmail =
-    objectForDetails && objectForDetails.email ? objectForDetails.email : 'Ingen email';
+  const contactEmail = profile && profile.email ? profile.email : 'Ingen email';
 
   //If we are looking at the details for a product, proposal or project, instead show the specific details for this
   if (props.productId) {
-    const productArray = useSelector((state) => state.products.availableProducts).filter(
-      (prod) => prod.id === props.productId
+    const productArray = useSelector((state) =>
+      state.products.availableProducts.filter((prod) => prod.id === props.productId)
     );
 
```

Comment:
Why do we use filter() here and find() above? Just curious :)

Reply:
I just moved the`.filter` inside the products array selector, this is the same behavior as the older commit. I'll refactor it on the next issue, to use `find` method. :)